### ISSUE TITLE
False positive for SC2086 when subshell

### ISSUE THREAD

**Issue by kkmuffme** [koalaman/shellcheck#1434]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
if [[ "$(dig +short $HOSTNAME)" == "$(dig +short example.com)" ]]
then
    echo "this is example.com"
fi
```

#### Here's what shellcheck currently says:
Double quote to prevent globbing and word splitting.

#### Here's what I wanted or expected to see:
no error at all

**Comment by austin987** [comment]
It's warning that $HOSTNAME needs to be quoted, not the subshell

**Comment by kkmuffme** [comment]
yeah but

**Comment by mattsawyer77** [comment]
@austin987 this is still problematic, though. for example:

```sh
if [[ -n "$(echo $possibly_empty_var)" ]]; then
   # do something
fi
```
is there a way to satisfy shellcheck here? maybe my bash skills are subpar and there is just a better way to do this in the code.

**Comment by Nightfirecat** [comment]
Quote `$possibly_empty_var` within the command substitution.

```bash
if [[ -n "$(echo "$possibly_empty_var")" ]]; then
   # do something
fi
```
